### PR TITLE
feat: Authentication

### PR DESCRIPTION
**Pull request by PeeyushPrashant**
- Login
- SignUp
- LogOut
- Toastify

Screenshot (90)
Screenshot (91)
Screenshot (92)

### CODE REVIEW

File: src/contexts/Auth-context.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        localStorage.setItem(
+          "auth",
+          JSON.stringify({ token: response.data.encodedToken })
+        );
+
+        setToken(response.data.encodedToken);
+        navigate("/products");
+      }
+    } catch (error) {
+      alert("Your account doesn't exist");
```
Comment (anuj-kumary) [inline src/contexts/Auth-context.js]: Why you're using here alert or everywhere Toastify

File: src/pages/Auth/SIgnUp.jsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    return (
+        <main className="main">
+        <div className="auth-container flex-row">
+          <div className="auth-card flex-col">
+            <h1 className="auth-heading">Sign Up</h1>
+            <p>
+              Fill below form to sign up and enjoy special offers in one8 store
+            </p>
+            <div className="auth-input flex-row">
+              <label for="" className="input-label"
```
Comment (anuj-kumary) [inline src/pages/Auth/SIgnUp.jsx:24]: Don't give keep empty label for

File: src/pages/Auth/SIgnUp.jsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+            
+            <div className="auth-input flex-row">
+              <label for="" className="input-label"><strong>Email</strong></label>
+              <input
+                type="text"
+                className="input-feild"
+                value={signUpForm.email}
+                placeholder="one8@gmail.com"
+                onChange={(e)=>setSignUpForm({...signUpForm,email:e.target.value})}
+                required
```
Comment (anuj-kumary) [inline src/pages/Auth/SIgnUp.jsx]: You can use type email for email input

**Review by anuj-kumary — APPROVED** [review]

**Review by Rohanmond — APPROVED** [review]
code is looking good to merge

**Pull request merged**